File: packages/api/aqapi/language/dao/language_dao.py
from typing import List, Optional
from sqlalchemy.ext.asyncio import AsyncSession
from aqapi.core.di_container import di_container
from aqapi.language.entity.languages_entity import LanguagesEntity
from aqapi.core.dao.base_dao import BaseDao
from aqapi.core.cache.redis_cacher import RedisCacher

class LanguageDao(BaseDao):
    """言語DAOクラス"""
    redis = di_container.get(RedisCacher)

    def __init__(self, session: AsyncSession):
        super().__init__(session)

    @property
    def entity_class(self) -> type[LanguagesEntity]:
        return LanguagesEntity
    
    @redis.cache("languages:all")
    async def fetch_all(self) -> List[LanguagesEntity]:
        return await super().fetch_all()

File: packages/backend/src/features/language/dao/languageDao.ts
import { EntityManager } from 'typeorm';
import { BaseDao } from '@backend/core/dao/baseDao';
import { LanguageEntity } from '../entity/languageEntity';

/**
 * 言語DAOクラス
 * 言語マスタデータのデータアクセスを管理する
 */
export class LanguageDao extends BaseDao<LanguageEntity> {
  constructor(session: EntityManager) {
    super(session);
  }

  protected get entityClass(): new () => LanguageEntity {
    return LanguageEntity;
  }
}
